arangod/Aql/EnumerateCollectionExecutor.h

```diff
@@ -126,20 +137,45 @@ class EnumerateCollectionExecutor {
   std::pair<ExecutionState, Stats> produceRows(OutputAqlItemRow& output);
   std::tuple<ExecutionState, EnumerateCollectionStats, size_t> skipRows(size_t atMost);
 
+  /**
+   * @brief produce the next Rows of Aql Values.
+   *
+   * @return ExecutorState, the stats, and a new Call that needs to be send to upstream
+   */
+  [[nodiscard]] std::tuple<ExecutorState, Stats, AqlCall> produceRows(
+      AqlItemBlockInputRange& input, OutputAqlItemRow& output);
+
+  uint64_t skipEntries(size_t toSkip, EnumerateCollectionStats& stats);
+  /**
+   * @brief skip the next Row of Aql Values.
+   *
+   * @return ExecutorState, the stats, and a new Call that needs to be send to upstream
+   */
+  [[nodiscard]] std::tuple<ExecutorState, Stats, size_t, AqlCall> skipRowsRange(
+      AqlItemBlockInputRange& inputRange, AqlCall& call);
+
   void initializeCursor();
 
  private:
   bool waitForSatellites(ExecutionEngine* engine, Collection const* collection) const;
 
+  void setAllowCoveringIndexOptimization(bool allowCoveringIndexOptimization);
+
+  /// @brief whether or not we are allowed to use the covering index
+  /// optimization in a callback
+  bool getAllowCoveringIndexOptimization() const noexcept;
+
  private:
   Infos& _infos;
   Fetcher& _fetcher;
```

Comment:
If you remove the old code, you need to get rid of this local reference as well.